src/ngraph/pass/graph_rewrite.cpp

```diff
@@ -139,41 +131,39 @@ bool pass::GraphRewrite::is_enabled(shared_ptr<pattern::Matcher> m)
     return true;
 }
 
-void pass::GraphRewrite::add_matcher(shared_ptr<pattern::Matcher> m)
+void pass::GraphRewrite::add_matcher(const shared_ptr<pattern::Matcher>& m,
+                                     const graph_rewrite_callback& callback)
 {
     if (is_enabled(m))
     {
-        m_matchers.push_back(m);
+        m_matchers.push_back({m, callback});
     }
 }
 
+void pass::RecurrentGraphRewrite::add_matcher(
+    const std::shared_ptr<pattern::RecurrentMatcher>& m,
+    const ngraph::recurrent_graph_rewrite_callback& callback)
+{
+    m_matchers.push_back({m, callback});
+}
+
 bool pass::RecurrentGraphRewrite::run_on_function(shared_ptr<Function> f)
 {
     bool changed = false;
     size_t i = 0;
-    bool is_dynamic_function = f->is_dynamic();
     do
     {
         for (auto node : f->get_ops())
         {
-            for (auto matcher : m_matchers)
+            for (auto& mc : m_matchers)
             {
-                if (is_dynamic_function &&
-                    (matcher->get_property(pass::PassProperty::REQUIRE_STATIC_SHAPE)))
```

Comment:
Do we want to keep the `REQUIRE_STATIC_SHAPE` logic? (Has it been moved elsewhere?)